e2e/libs/utility/utility.py

```diff
@@ -285,3 +285,25 @@ def get_name_suffix(*args):
         if arg:
             suffix += f"-{arg}"
     return suffix
+
+
+def check_popen_process_not_completed(process):
```

Comment:
Is it necessary to introduce `popen`, `process.poll()`, `process.terminate()` and `process.communicate()`?

Can `subporcess.check_output()` with the `timeout` parameter accomplish the same goal?

ref: https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module

Reply:
Yes, in test case `Setting Allow Node Drain with the Last Healthy Replica protects the last healthy replica with Pod Disruption Budget (PDB)`,  fter change `node-drain-policy` from `block-if-contains-last-replica` to `always-allow`, it 
 need to validate if the drain process completed or not, currently I can not find a way to let `subporcess.check_output()` execute in background. that why I choosed `popen`

Reply:
Is the following approach feasible?

```
def force_drain_node():
    subprocess.check_output(timeout=90)
```

```
Force drain node ${node_id} and expect failure
    Run Keyword And Expect Failure    force drain node
```

```
Force drain node ${node_id} and expect success
    force drain node
```

```
And Set setting node-drain-policy to block-if-contains-last-replica
...
...
When Force drain node 2 and expect fail
And Check instance-manager pod is running on node 2

When Set setting node-drain-policy to always-allow
And Force drain node 2 and expect success
And Check PDB not exist    
```

Reply:
Hi @yangchiu, updated and test [result](https://ci.longhorn.io/job/private/job/longhorn-e2e-test/1492/)